src/options/options.js

```diff
@@ -171,6 +207,84 @@ document.addEventListener("alpine:init", () => {
           "Saved successfully, you can close this page.";
       });
     },
+    async startOAuthFlow() {
+      let codeVerifier = generateCodeVerifier();
+      let codeChallenge = await generateCodeChallenge(codeVerifier);
+
+      let authUrl = new URL(`${ORBIT_API_ROOT_URL}/oauth/authorize`);
+
+      let params = new URLSearchParams({
+        client_id: OAUTH_CLIENT_ID,
+        response_type: "code",
+        scopes: "read write",
+        code_challenge_method: "S256",
+        redirect_uri: chrome.identity.getRedirectURL("oauth2"),
+        code_challenge: codeChallenge,
+      });
+
+      authUrl.search = params.toString();
+
+      chrome.identity.launchWebAuthFlow(
+        { url: authUrl.toString(), interactive: true },
+        function (redirectUrl) {
+          if (redirectUrl) {
+            var parsed = parseQueryParams(redirectUrl);
+
+            return this.getOAuthToken(parsed.code, codeVerifier);
+          } else {
+            // TODO: Verify failure routes from OAuth
+            console.error(
+              "launchWebAuthFlow login failed. Is your redirect URL (" +
+                chrome.identity.getRedirectURL("oauth2") +
+                ") configured with your OAuth2 provider?"
+            );
+            return null;
+          }
+        }.bind(this)
+      );
+    },
+    async getOAuthToken(oAuthCode, codeVerifier) {
+      let authUrl = new URL(`${ORBIT_API_ROOT_URL}/oauth/token`);
+
+      let params = new URLSearchParams({
+        client_id: OAUTH_CLIENT_ID,
+        grant_type: "authorization_code",
+        code: oAuthCode,
+        code_verifier: codeVerifier,
+        redirect_uri: chrome.identity.getRedirectURL("oauth2"),
+      });
+
+      authUrl.search = params.toString();
+
+      try {
+        const response = await fetch(authUrl.toString(), {
+          method: "POST",
+        });
+
+        const { access_token, refresh_token, expires_in } =
+          await response.json();
+
+        // Calculate timestamp when OAuth token expires - current time + it's expires_in timestamp
+        const expiresAt = Math.floor(Date.now() / 1000) + expires_in;
```

Comment:
Is there a benefit to storing `expiresAt` (a computed property), rather than storing `expiresIn` (the exact response from the API) and changing how we check expiration in `_isOAuthTokenExpired`?

Reply:
If we store `expiresIn` we'd also need to store `tokenSetAt` (or similar - ie, the timestamp when we received the token). Then we'd check `storage.expiresIn + storage.setAt < current_time`. 

The benefit of the current approach is we're only storing one value. Happy to switch, feels like a minor saving if any - WDYT?

Reply:
Oooooh of course! I had totally missed that. Thanks!